formats/keep-json.ts

```diff
@@ -0,0 +1,202 @@
+			.setDesc('If imported, files archived in Google Keep will be tagged as archived.')
+            .addToggle(toggle => {
+                toggle.setValue(this.importArchived)
+                toggle.onChange(async (value) => {
+                    this.importArchived = value;
+                });
+            });
+        
+		this.importTrashedSetting = new Setting(this.modal.contentEl)
+            .setName('Import deleted notes')
+			.setDesc('If imported, files deleted in Google Keep will be tagged as deleted. Deleted notes will only exist in your Google export if deleted recently.')
+            .addToggle(toggle => {
+                toggle.setValue(this.importTrashed)
+                toggle.onChange(async (value) => {
+                    this.importTrashed = value;
+                });
+            });
+
+		this.addOutputLocationSetting('Google Keep');
+
+	}
+
+	async import(progress: ProgressReporter): Promise<void> {
+		let { files } = this;
+
+		if (files.length === 0) {
+			new Notice('Please pick at least one file to import.');
+			return;
+		}
+
+		let folder = await this.getOutputFolder();
+		if (!folder) {
+			new Notice('Please select a location to import your files to.');
+			return;
+		}
+		let assetFolderPath = `${folder.path}/Assets`;
+
+		for (let file of files) {
+			try {
+				if(file.extension === 'zip') {
+					await this.readZipEntries(file, folder, assetFolderPath, progress);
+					
+				} else if(file.extension === 'json') {
+					let rawContent = await file.readText();
+					await this.importKeepNote(rawContent, folder, file.basename, progress);
+
+				} else {
+					const arrayBuffer = await file.read();
+					await this.copyFile(arrayBuffer, assetFolderPath, file.name, progress);
+				}
+
+			} catch (e) {
+				console.error(`${file.name} ::: `, e);
+				progress.reportFailed(file.name, e);
+			}
+		}
+
+	}
+
+	async readZipEntries(file: PickedFile, folder: TFolder, assetFolderPath: string, progress: ProgressReporter) {
+		await file.readZip(async zip => {
+			for (let entry of await zip.getEntries()) {
+				if (!entry || entry.directory || !entry.getData) return;
+				let curInnerFilename = '';
+				try {
+					let innerFileProps = parseFilePath(entry.filename);
+					curInnerFilename = innerFileProps.name;
+		
+					if(innerFileProps.extension === 'json') {
+						let rawContent = await entry.getData(new TextWriter());
+						await this.importKeepNote(rawContent, folder, innerFileProps.basename, progress);
+		
+					} else if(ATTACHMENT_EXTS.contains(innerFileProps.extension)) {
+						const rawContent = await entry.getData(new BlobWriter());
+						const arrayBuffer = await rawContent.arrayBuffer();
+						await this.copyFile(arrayBuffer, assetFolderPath, innerFileProps.name, progress);
+					}
+					// else: Silently skip any other unsupported files in the zip
+		
+				} catch(e) {
+					console.error(`${file.name} ::: ${curInnerFilename ?? curInnerFilename}`, e);
```

Comment:
Let's avoid adding extra console messages since they are already reported in `reportFailed`.

Reply:
Fixed and Pushed.